ros_buildfarm/status_page.py

```diff
@@ -873,6 +928,98 @@ def _get_blocked_releases_info(config_url, rosdistro_name, repo_names=None):
     return repos_info
 
 
+def _has_source_entry(repo_name, dist_file):
+    return repo_name in dist_file.repositories and \
+        dist_file.repositories[repo_name].source_repository is not None and \
+        dist_file.repositories[repo_name].source_repository.url is not None
+
+
+def _prev_rosdistro(index, rosdistro_name):
+    """Given current rosdistro name, return the previous."""
+    valid_rosdistro_names = list(index.distributions.keys())
+    valid_rosdistro_names.sort()
+
+    # skip distributions with a different type if the information is available
+    distro_type = index.distributions[rosdistro_name].get('distribution_type')
+    if distro_type is not None:
+        valid_rosdistro_names = [
+            n for n in valid_rosdistro_names
+            if distro_type == index.distributions[n].get('distribution_type')]
+
+    try:
+        i = valid_rosdistro_names.index(rosdistro_name)
+    except ValueError:
+        raise ValueError('Distribution key not found in list of valid distributions.')
+
+    if i == 0:
+        raise ValueError('No previous distribution found.')
+
+    return valid_rosdistro_names[i - 1]
+
+
+def _rosdistro_or_latest(index, rosdistro_name=None):
```

Comment:
I don't see where this function is ever used with a `rosdistro_name` being `None`?

Reply:
Hmm I don't either. I copied it from here, but I'll remove that altogether. https://github.com/ros-infrastructure/ros_buildfarm/blob/467c34fb812281c32307c6b6339c2da13d394907/ros_buildfarm/status_page.py#L695-L698

Reply:
Removed `_rosdistro_or_latest()` in 5c2b1f3